### ISSUE TITLE
Add debug command

### ISSUE THREAD

**Issue by daeyeon** [Samsung/iotjscode#87]
I think that the following features will improve debugging experience of iotjs code.

- Required Features:
Continue, Step-Over, Step-In, Step-Out,  Restart, Stop for Debugging

- Current Status:
We have three buttons for continue/stop, step-in, and next for debug status. `next` button naming is a bit confusing compared with other debuggers.

**Comment by knightburton** [comment]
@daeyeon I agree. Would you describe that to me what does the restart exactly mean in our case? This would be the context reset or something else?

**Comment by daeyeon** [comment]
@knightburton It's similar to the context reset. My observation is that the context reset is available only when a remote server is started with waiting sources. Is that right? If so, it needs to be available also when IoT.JS Code directly uses sources from the remote server.

In other words, it is meant that literally restart debugging from the beginning, but breakpoints, loaded sources, and watch variables remain. So users can mainly skip the steps to load sources from a remote server. Because what user want to load are already done.

The following is a user scenario expected.

1. There is only one break point in any `for-loop` block and user stays at the point in IoT.JS Code. 
2. He is investigating changes using debugger `Next`. The iterator of `for-loop` is increased.
3. At any point, he wants to watch it from the beginning of the iterator.
4. He pushes `Restart`.
5. In user perspective, he will be at the breakpoint (mentioned in step 1) again without any extra steps. The iterator points to its beginning. 

Hopefully, this can help your understanding.

**Comment by knightburton** [comment]
Yes, thank you for your clarify. These new feature are require changes in jerryscript, too.

**Comment by knightburton** [comment]
@daeyeon I have a question about the last requested command, the reset command. Do we want to support the reset both in normal and wait-source mode? Because of the current implementation in jerryscript is only allows context reset in source waiting mode which is enabled when the engine was started in that mode and gets the sources on-the-fly from the iotjscode. Besides, in normal mode when the engine gets the sources from command line, in this case the source waiting and the context reset is not allowed.

So, should we change the design of this in the jerry or just allow the reset command in source waiting mode?
